### PR TITLE
add example with dynamic group of progress bars

### PR DESCRIPTION
**Pull request by boegel**
## Type of changes

- [ ] Bug fix
- [ ] New feature
- [x] Documentation / docstrings
- [ ] Tests
- [x] Other: examples

## Checklist

- [x] I've run the latest [black](https://github.com/psf/black) with default args on new code.
- [x] I've updated CHANGELOG.md and CONTRIBUTORS.md where appropriate.
- [ ] I've added tests for new code.
   - Is this required for examples too? If so, happy to look into it...
- [x] I accept that @willmcgugan may be pedantic in the code review.

## Description

Additional example for progress bars, more dynamic in nature than the existing [live_progress.py](https://github.com/willmcgugan/rich/blob/master/examples/live_progress.py>); see also discussion in #1500 .

Screenshot while running:

image

Screenshot when done:

image

### CODE REVIEW

File: examples/dynamic_progress.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+Demonstrates how to create a dynamic group of progress bars,
+showing multi-level progress for multiple tasks (installing apps in the example),
+each of which consisting of multiple steps.
+
+"""
+
+import time
+
+from rich.console import RenderGroup
```
Comment (willmcgugan) [inline examples/dynamic_progress.py:11]: RenderGroup was deprecate, the new name is just `Group`

File: examples/dynamic_progress.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+)
+# overall progress bar
+overall_progress = Progress(
+    TimeElapsedColumn(),
+    BarColumn(),
+    TextColumn("{task.description}")
+)
+# group of progress bars;
+# some are always visible, others will disappear when progress is complete
+group = RenderGroup(
```
Comment (willmcgugan) [inline examples/dynamic_progress.py:65]: Should be `Group` here. Could you also give `group` a more descriptive name?

**Review by willmcgugan — CHANGES_REQUESTED** [review]
Looks good! Couple of small requests.

Could you run this against the latest black?

**Comment by willmcgugan** [comment]
Nice work!

Could you maybe put the app progress in a Panel to separate it visually? And speed it up a fraction...

**Comment by boegel** [comment]
@willmcgugan Anything in particular that's blocking this PR? Happy to make additional changes if needed.

**Review by willmcgugan — APPROVED** [review]

**Comment by willmcgugan** [comment]
Sorry, forgot about this one. Looks good. Thanks!

**Pull request merged**